tests/qmctl-test/scripts/qmctl_cp_bidirectional.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../e2e/lib/utils
+
+QMCTL_SCRIPT="../../tools/qmctl/qmctl"
+TEST_RESULTS=0
+
+info_message "Starting comprehensive qmctl copy tests..."
+
+# Check if container is available
+if ! python3 "$QMCTL_SCRIPT" exec echo "container_check" > /dev/null 2>&1; then
+    fail_message "QM container is not available - copy tests cannot run"
+    echo "Please ensure QM container is running with: sudo systemctl start qm"
+    exit 1
+fi
+
+info_message "QM container is available - running comprehensive copy tests"
+
+# Test 0: Basic copy test (from original qmctl_cp.sh)
+info_message "Testing: Basic copy test (host to container)"
+python3 "$QMCTL_SCRIPT" cp ./files/file-to-copy.txt qm:/tmp
```

Comment:
where this file file-to-copy.txt is created?

Reply:
I am not sure I understand this question correctly.
file-to-copy.txt is located and already created in qmctl-test/files
, and it is copied to the qm container in /tmp